app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,27 @@
+class PasswordResetsController < ApplicationController
+  def show
+    user = User.where(token: params[:id]).take
+    if user
+      @token = user.token
+    else
+      redirect_to expired_token_path
+    end
+  end
+
+  def create
+    user = User.where(token: params[:token]).take
+    if user
+      user.password = params[:password]
+      user.generate_token
```

Comment:
I would suggest to remove the token here and not generate another one, because as you said it is a security threat and makes encrypting password useless.